Shared/Samples/Show viewshed from geoelement in scene/ShowViewshedFromGeoelementInSceneView.swift

```diff
@@ -0,0 +1,200 @@
+                .background(.thinMaterial, ignoresSafeAreaEdges: .horizontal)
+        }
+    }
+}
+
+private extension ShowViewshedFromGeoelementInSceneView {
+    /// The view model for the sample.
+    class Model: ObservableObject {
+        /// A scene with an imagery basemap.
+        let scene: ArcGIS.Scene = {
+            let scene = Scene(basemapStyle: .arcGISImagery)
+            
+            // Add elevation source to the base surface of the scene with the service URL.
+            let elevationSource = ArcGISTiledElevationSource(url: .worldElevationService)
+            scene.baseSurface.addElevationSource(elevationSource)
+            
+            // Create the building layer and add it to the scene.
+            let buildingsLayer = ArcGISSceneLayer(url: .brestBuildingsService)
+            scene.addOperationalLayer(buildingsLayer)
+            
+            // Set scene the viewpoint specified by the camera position.
+            let point = Point(x: -73.0870, y: -49.3460, z: 5046, spatialReference: .wgs84)
+            let camera = Camera(location: point, heading: 11, pitch: 62, roll: 0)
+            scene.initialViewpoint = Viewpoint(boundingGeometry: point, camera: camera)
+            
+            return scene
+        }()
+        
+        /// A camera controller set to follow the tank.
+        let cameraController: CameraController
+        
+        /// The graphics overlay for the tank graphic.
+        let graphicsOverlay: GraphicsOverlay = {
+            let graphicsOverlay = GraphicsOverlay()
+            graphicsOverlay.sceneProperties = LayerSceneProperties(surfacePlacement: .relative)
+            
+            // Set up the heading expression for the tank.
+            let renderer3D = SimpleRenderer()
+            let sceneProperties = RendererSceneProperties(
+                headingExpression: "[heading] + 90",
+                pitchExpression: "[pitch]",
+                rollExpression: "[roll]"
+            )
+            sceneProperties.headingExpression = "[HEADING]"
+            renderer3D.sceneProperties = sceneProperties
+            graphicsOverlay.renderer = renderer3D
+            
+            return graphicsOverlay
+        }()
+        
+        /// The analysis overlay for the tank's viewshed.
+        let analysisOverlay = AnalysisOverlay()
+        
+        /// The graphic for the tank.
+        let tankGraphic: Graphic = {
+            let tankSymbol = ModelSceneSymbol(url: .bradleyTank, scale: 10)
+            tankSymbol.heading = 90
+            tankSymbol.anchorPosition = .bottom
+            let tankGraphic = Graphic(
+                geometry: Point(x: -4.506390, y: 48.385624, spatialReference: .wgs84),
+                attributes: ["HEADING": 0.0],
+                symbol: tankSymbol
+            )
+            return tankGraphic
+        }()
+        
+        /// The point for the tank to move toward.
+        var waypoint: Point!
```

Comment:
Might still be a good idea to guard let the optional waypoint, or even better, only allow 1 animation to run at once. Consider this case: If you consecutively tap on the map twice at 2 different points, when the first waypoint is nullified, the second scheduled animate call will crash.